domain/modelconfig/state/state.go

```diff
@@ -244,3 +244,25 @@ WHERE key = excluded.key
 		return nil
 	})
 }
+
+// CheckSpace checks if the space identified by the given space name exists.
+func (st *State) CheckSpace(ctx context.Context, spaceName string) (bool, error) {
```

Comment:
I think this should be renamed to something like `SpaceExists`

Reply:
Yes `SpaceExists` is better.

Reply:
Sure! fixed.